### ISSUE TITLE
Jose Library Not working with Bun

### ISSUE THREAD

**Issue by Lillious** [oven-sh/bun#1283]
### Version

v0.1.13

### Platform

Darwin [redacted] 21.6.0 Darwin Kernel Version 21.6.0: Wed Sept 28 11:07:10 EDT 2022; root:xnu-8020.140.49~2/RELEASE_X86_64 x86_64

### What steps will reproduce the bug?

**const { ExpressOIDC } = require('@okta/oidc-middleware');**

### Requiring a package that has Jose listed as a dependency results in the following error:

```
[28/Sep/2022:14:55:48] 2 |
[28/Sep/2022:14:55:48] 3 | const { name: secp256k1 } = require('../jwk/key/secp256k1_crv')
[28/Sep/2022:14:55:48] 4 |
[28/Sep/2022:14:55:48] 5 | const curves = new Set()
[28/Sep/2022:14:55:48] 6 |
[28/Sep/2022:14:55:48] 7 | if (getCurves().includes('prime256v1')) {
[28/Sep/2022:14:55:48] ^
[28/Sep/2022:14:55:48] TypeError: getCurves is not a function. (In 'getCurves()', 'getCurves' is undefined)
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/node_modules/jose/lib/registry/ec_curves.js:7:4
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/node_modules/jose/lib/registry/index.js:1:26
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/node_modules/jose/lib/help/generate_iv.js:3:30
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/node_modules/jose/lib/jwe/encrypt.js:4:27
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/node_modules/jose/lib/jwe/index.js:1:24
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/node_modules/jose/lib/index.js:2:15
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/node_modules/openid-client/lib/issuer.js:6:21
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/node_modules/openid-client/lib/index.js:1:23
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/src/oidcUtil.js:14:29
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/src/ExpressOIDC.js:15:25
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/node_modules/@okta/oidc-middleware/index.js:14:37
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
[28/Sep/2022:14:55:48] at /usr/src/app/app.js:6:32
[28/Sep/2022:14:55:48] at bun:wrap:1:16360
```

### How often does it reproduce? Is there a required condition?

Every time, causes app to crash.

### What is the expected behavior?

Bun should be able to successfully load and run Jose as it's mentioned as compatible.
This currently works in the node environment; however, upon switching runtimes and runtimes only, I was presented with this error.

### What do you see instead?

The above error.

**Comment by Jarred-Sumner** [comment]
Jose should be working as of Bun v0.2.2 since Web Crypto API is included. Can you confirm if this is still happening?

**Comment by Electroid** [comment]
We're working on a fix, tracking this here: #1448

**Comment by panva** [comment]
@Lillious has `jose` version 1 or 2 as a transitive dependency, these early versions of jose are only for the Node.js runtime, no amount of unmaintained and outdated polyfilling (i.e. what bun passes as node's crypto using crypto-browserify) will help.

Neither will #1448 and presence of webcrypto.

Updating `@okta/oidc-middleware` to v5 could since it depends on new openid-client which then depends on new jose, but then the question is still node compat parity for everything but jose...
